src/test/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryServiceTest.java

```diff
@@ -628,4 +629,112 @@ public void testCalculateRunningBalance_GroupedByAgencyAndAccountType() {
                 .containsExactlyInAnyOrder(200L, "LEI", "SPND");
         }
     }
+
+    @Nested
+    public class RelatedTransactionRunningBalance {
+        @Test
+        public void testCalculateRunningBalanceOverMultipleDays() {
+            final var yesterdaysDate = LocalDateTime.now().minusDays(1);
```

Comment:
Trying to make it understandable by giving the TXs names - maybe better to refer to day1 and day2 or something?